src/Compilers/CSharp/Portable/Binder/Binder_AnonymousTypes.cs

```diff
@@ -40,8 +40,8 @@ private BoundExpression BindAnonymousObjectCreation(AnonymousObjectCreationExpre
                 AnonymousObjectMemberDeclaratorSyntax fieldInitializer = initializers[i];
                 NameEqualsSyntax nameEquals = fieldInitializer.NameEquals;
                 ExpressionSyntax expression = fieldInitializer.Expression;
-
-                SyntaxToken nameToken = default(SyntaxToken);
+                _ = default(SyntaxToken);
```

Comment:
this just seems bad not good.

Reply:
We can extend the analyzer to mark default expressions as expressions with no side effects and the fixer will correctly remove this whole statement.

Reply:
👍   I think that would be good.

Reply:
Fixed in https://github.com/dotnet/roslyn/pull/30777/commits/1182f80ac247fe876013d5baa0a27d02f9eaab08